frontend/src/pages/device/index.vue

```diff
@@ -10,27 +10,58 @@
-        <SectionTopMenu>
-            <template #hero>
-                <div class="flex-grow space-x-6 items-center inline-flex">
-                    <router-link :to="navigation[0]?navigation[0].path:''" class="inline-flex items-center" data-nav="device-overview">
-                        <div class="text-gray-800 text-xl font-bold">{{ device?.name }}</div>
-                        <div class="text-gray-400 text-md font-bold ml-3">{{ device?.type }}</div>
-                    </router-link>
-                </div>
-            </template>
-        </SectionTopMenu>
+        <div class="ff-instance-header">
+            <InstanceStatusHeader>
+                <template #hero>
+                    <div class="flex-grow items-center inline-flex flex-wrap" data-el="device-name">
+                        <div class="text-gray-800 text-xl font-bold mr-6">
+                            {{ device.name }}
+                        </div>
+                        <DeviceLastSeenBadge class="mr-6" :last-seen-at="device.lastSeenAt" :last-seen-ms="device.lastSeenMs" :last-seen-since="device.lastSeenSince" />
+                        <ProjectStatusBadge :status="device.status"/>
+                        <div class="w-full text-sm mt-1">
+                            <div v-if="device?.project">
+                                Instance:
+                                <router-link :to="{name: 'Instance', params: {id: device.project.id}}" class="text-blue-600 cursor-pointer hover:text-blue-700 hover:underline">{{ device.project.name }}</router-link>
+                            </div>
+                            <span v-else class="text-gray-400 italic">Device Not Assigned to an Instance</span>
+                        </div>
+                    </div>
+                </template>
+                <template #tools>
+                    <div class="space-x-2 flex align-center">
+                        <div v-if="editorAvailable">
+                            <a v-if="!isVisitingAdmin" :href="instance.url" target="_blank" class="ff-btn ff-btn--secondary" data-action="open-editor">
+                                Open Editor
+                                <span class="ff-btn--icon ff-btn--icon-right">
+                                    <ExternalLinkIcon />
+                                </span>
+                            </a>
+                            <button v-else title="Unable to open editor when visiting as an admin" class="ff-btn ff-btn--secondary" disabled>
+                                Open Editor
+                                <span class="ff-btn--icon ff-btn--icon-right">
+                                    <ExternalLinkIcon />
+                                </span>
+                            </button>
+                        </div>
+                        <DropdownMenu v-if="hasPermission('project:change-status')" buttonClass="ff-btn ff-btn--primary" :options="actionsDropdownOptions">Actions</DropdownMenu>
```

Comment:
`actionsDropdownOptions` isn't defined on this page

Reply:
Removed this anyway, as wasn't meant to be included (lazy copy + paste)